junit-test/shared/src/test/scala/org/scalajs/junit/utils/JUnitTest.scala

```diff
@@ -236,21 +240,31 @@ abstract class JUnitTest {
 object JUnitTest {
   sealed trait Output
   final case class Log(level: Char, msg: String) extends Output
-  final case class Event(status: Status, testName: String) extends Output
+  final case class Event(status: Status, testName: String, throwableToString: Option[String]) extends Output
   final case class Done(msg: String) extends Output
 
   object Output {
+    // We need something that does not occur in test names nor in Throwable.toString.
+    val separator: String = "::"
+
     def deserialize(line: String): Output = line.toList match {
       case 'l' :: level :: msg  => Log(level, msg.mkString(""))
-      case 'e' :: s :: testName => Event(Status.values()(s - '0'), testName.mkString(""))
       case 'd' :: msg           => Done(msg.mkString(""))
+      case 'e' :: s :: tail =>
```

Comment:
Coding style nit. There should be blank line between those two lines.

Reply:
done